docs/resources/flex_cluster.md

```diff
@@ -0,0 +1,101 @@
+# Resource: mongodbatlas_flex_cluster
+
+`mongodbatlas_flex_cluster` provides a Flex Cluster resource. The resource lets you create, update, delete and import a flex cluster.
+
+-> **NOTE**: Flex Clusters are currently in Preview. To use this feature, you must take the following actions:
+1. Enable the `Atlas Uss Enabled` Preview Feature in your project (contact [MongoDB Support](https://www.mongodb.com/services/support)).
+2. Enable the [Preview Features](https://github.com/mongodb/terraform-provider-mongodbatlas?tab=readme-ov-file#preview-features) when running `terraform` commands.
+
+## Example Usages
+
+```terraform
+resource "mongodbatlas_flex_cluster" "flex_cluster" {
+  project_id = var.project_id
+  name       = var.cluster_name
+  provider_settings = {
+    backing_provider_name = "AWS"
+    region_name           = "US_EAST_1"
+  }
+  termination_protection_enabled = true
+}
+
+data "mongodbatlas_flex_cluster" "flex_cluster" {
+  project_id = var.project_id
+  name       = mongodbatlas_flex_cluster.flex_cluster.name
+}
+
+data "mongodbatlas_flex_clusters" "flex_clusters" {
+  project_id = var.project_id
+}
+
+output "mongodbatlas_flex_cluster" {
+  value = data.mongodbatlas_flex_cluster.flex_cluster.name
+}
+
+output "mongodbatlas_flex_clusters_names" {
+  value = [for cluster in data.mongodbatlas_flex_clusters.flex_clusters.results : cluster.name]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Human-readable label that identifies the instance.
+- `project_id` (String) Unique 24-hexadecimal character string that identifies the project.
+- `provider_settings` (Attributes) Group of cloud provider settings that configure the provisioned MongoDB flex cluster. (see [below for nested schema](#nestedatt--provider_settings))
+
+### Optional
+
+- `tags` (Map of String) Map that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the instance.
+- `termination_protection_enabled` (Boolean) Flag that indicates whether termination protection is enabled on the cluster. If set to `true`, MongoDB Cloud won't delete the cluster. If set to `false`, MongoDB Cloud will delete the cluster.
+
+### Read-Only
+
+- `backup_settings` (Attributes) Flex backup configuration (see [below for nested schema](#nestedatt--backup_settings))
+- `cluster_type` (String) Flex cluster topology.
```

Comment:
Do we have any more details about this?

Reply:
This is what was generated from the [API](https://github.com/search?q=repo%3A10gen%2Fmms+%22Flex+cluster+topology%22&type=code).